packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -252,7 +331,35 @@ export function useConnections({
         });
       }
     },
-    [activeConnectionId, deleteConnection, dispatch]
+    [
+      activeConnectionId,
+      repositoryRemoveConnection,
+      dispatch,
+      getActiveAndInactiveCount,
+      onConnectionRemoved,
+      connectionsManager,
+    ]
+  );
+
+  const saveConnection = useCallback(
+    async (connectionInfo: ConnectionInfo) => {
+      const saved = await saveConnectionInfo({
+        fullConnectionInfo: connectionInfo,
+      });
+      if (!saved) {
+        return;
+      }
+
+      if (activeConnectionId === connectionInfo.id) {
+        // Update the active connection if it's currently selected.
+        dispatch({
```

Comment:
Why other places that save info directly through saveConnectionInfo don't need to update this?

Reply:
good question. I actually only moved this function and I haven't questioned this bit. I'd say this is a legacy thing, because in multiple connections the activeConnection is only used by the connection form, and the form is closed when we save. perhaps @himanshusinghs can confirm? if I'm right we could mark this for a cleanup